File: java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java

package com.adaptionsoft.games.trivia.runner;
import java.util.ArrayList;
import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Game;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
		Game aGame = new Game();
		
		aGame.add("Chet");
		aGame.add("Pat");
		aGame.add("Sue");
		
		Random rand = new Random();

		var plays = new ArrayList<Play>(){};

		do {
			var roll = rand.nextInt(5) + 1;
			var isIncorrect = rand.nextInt(9) == 7;

			var play = new Play(roll, !isIncorrect);

			plays.add(play);

			aGame.roll(roll);



			if (isIncorrect) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.wasCorrectlyAnswered();
			}
			
			
			
		} while (notAWinner);

		for(Play play: plays){
			System.out.println(play.roll()+","+play.isCorrect());
		}
	}

	public record Play(int roll, boolean isCorrect) { }
}

File: C#/Trivia/Trivia/GameRunner.cs
using System;
using System.Collections.Generic;

namespace Trivia
{
    public class GameRunner
    {
        private static bool _notAWinner;

        public static void Main(string[] args)
        {
            var aGame = new Game();

            aGame.Add("Chet");
            aGame.Add("Pat");
            aGame.Add("Sue");

            var rand = new Random();

            var plays = new List<string>();

            do
            {
                var roll = rand.Next(5) + 1;
                aGame.Roll(roll);

                var isFail = rand.Next(2) == 1;
                if (isFail)
                {
                    _notAWinner = aGame.WrongAnswer();
                }
                else
                {
                    _notAWinner = aGame.WasCorrectlyAnswered();
                }
                plays.Add($"{roll},{!isFail}");
            } while (_notAWinner);

            foreach (var play in plays)
            {
                Console.WriteLine(play);
            }
        }
    }
}
